packages/explat-client-react-helpers/src/index.tsx

```diff
@@ -64,18 +64,18 @@ export default function createExPlatClientReactHelpers(
 			...providedOptions,
 		};
 
+		// Manual updates to ensure rerendering when we want it:
+		// https://legacy.reactjs.org/docs/hooks-faq.html#is-there-something-like-forceupdate
+		// eslint-disable-next-line @typescript-eslint/no-unused-vars
+		const [ _ignored, forceUpdate ] = useReducer( ( x ) => x + 1, 0 );
 		const [ previousExperimentName ] = useState( experimentName );
```

Comment:
Not related to this PR at all but you might want to consider using `useRef` for this since it doesn't really need the features of `useState`.